control-plane/csi-driver/src/bin/controller/client.rs

```diff
@@ -408,6 +408,16 @@ impl IoEngineApiClient {
             )
             .await?;
 
-        Ok(snapshots.into_body())
+        let next_token = snapshots.body().next_token;
+        // Return only the snapshots that are Created, leaving out other statuses if any.
+        Ok(models::VolumeSnapshots {
+            entries: snapshots
+                .into_body()
+                .entries
+                .into_iter()
+                .filter(|s| s.definition.metadata.status == models::SpecStatus::Created)
```

Comment:
hmm, perhaps it should instead filter out Creating state, otherwise we won't list snapshots in the delete process? Guess depends what behaviour we want here CC @Abhinandan-Purkait

Reply:
I think it might be confusing to list the ones in delete process because this status isn't sent back to CSI, so consumer won't know it's a Created one or Deleting one.

Reply:
tweaked the filter, please take a look.

Reply:
@tiagolobocastro Another thought around this - if we are listing a 'deleting' one, and the consumer picks that one up for next steps e.g. creating a clone, the clone create will fail possibly failing an application. There'll always be more successful deletions compared to sporadic deletion failures.
I'd say we don't want that kind of angst?

Reply:
If CSI has already tried to delete it it's because the CR has been deleted.